rust-runtime/protocol-test-helpers/src/lib.rs

```diff
@@ -310,11 +321,58 @@ fn try_json_eq(actual: &str, expected: &str) -> Result<(), ProtocolTestFailure>
     }
 }
 
+fn rewrite_url_encoded_map_keys(input: &str) -> (String, String) {
```

Comment:
since this file is already kinda mega, can you put this in its own module like XML?

Reply:
Definitely. Moved.